crates/runtime/src/dataconnector/graphql.rs

```diff
@@ -219,13 +269,168 @@ impl GraphQLClient {
             res.extend(batch);
         }
 
-        Ok((vec![res], schema))
+        Ok((res, schema, next_cursor))
+    }
+
+    async fn execute_paginated(
+        &self,
+        schema: SchemaRef,
+        limit: Option<usize>,
+    ) -> Result<Vec<Vec<RecordBatch>>> {
+        let (first_batch, _, mut next_cursor) =
+            self.execute(Some(Arc::clone(&schema)), limit, None).await?;
+        let mut res = vec![first_batch];
+
+        while let Some(next_cursor_val) = next_cursor {
+            let (next_batch, _, new_cursor) = self
+                .execute(
+                    Some(Arc::clone(&schema)),
```

Comment:
Let's move schema cloning outside of the loop

Reply:
Don't quite understand. We can't do it with arc currently.

Reply:
Ah, you're right. It is not a reference here